### PR TITLE
fix npm existing pkg checks

### PR DESCRIPTION
**Pull request by cozmo**
npm publishing had some...rough edges. Specifically 

With npm@2 builds would fail if you didn't bump the version every time. We were never hitting the `else` in the below block. 

With npm@3 builds would fail every time, always hitting the second part of the if statement, never the first. 

This happens because in v2 the `npm info` command returns `` and in v3 it returns `undefined`. 

For some reason the `-z` check wasn't correctly catching `` for v2, and obviously wasn't catching `undefined` for v3. This updates it to be slightly more explicit, and I've tested it with both versions to ensure it works. 

The build failed and not really sure what I'm doing so...would love some help with that. Also happy to take any sort of crazy bash optimization advice :p

### CODE REVIEW

**Comment by mohit** [comment]
I think this is because of Raf's slimming of the drone base image. Try removing the ruby stuff from the .drone.yml?

**Comment by cozmo** [comment]
@mohit Good news - seems like removing the ruby stuff fixed the build...bad news - had to remove some cryptic stuff from the Makefile that I don't _think_ we need but...no idea how to be sure.

**Comment by mohit** [comment]
luckily we don't need that stuff. The dockerfile builds the go binary: https://github.com/Clever/drone/blob/master/Dockerfile#L11-L15

**Comment by mohit** [comment]
did you test the npm publish through drone?

**Comment by mohit** [comment]
if tested then this looks good to me. Would be bad if this breaks all `npm publish`. Unless that's already the case.

**Comment by cozmo** [comment]
@mohit I tested it locally by running the command - I'm reasonably okay calling that "good enough" since it seems like the cost of rolling this out/back is fairly low - thoughts?

File: plugin/publish/npm/npm.go [hunk 1]
```diff
@@ -22,14 +22,16 @@ EOF
 const CmdPublish = `
 _NPM_PACKAGE_NAME=$(cd %s && npm list | head -n 1 | cut -d ' ' -f1)
 _NPM_PACKAGE_TAG="%s"
-if [ -z "$(npm info ${_NPM_PACKAGE_NAME} 2> /dev/null)" ]
+_EXISTING_NPM_PKG=$(npm info ${_NPM_PACKAGE_NAME} 2> /dev/null)
+if [ "${_EXISTING_NPM_PKG}" == "" ] || "${_EXISTING_NPM_PKG}" = "undefined" ]
```
Comment (peggyl) [inline plugin/publish/npm/npm.go]: Should it be `"${_EXISTING_NPM_PKG}" == "undefined"` with 2 equals signs?

Comment (jiancheung) [inline plugin/publish/npm/npm.go]: TIL: apparently in bash `==` and `=` are the same. both meant for string comparisons

**Pull request merged**